### PR TITLE
journal-remote: remove MHD_USE_PEDANTIC_CHECKS from the default flags

### PR DESCRIPTION
**Pull request by yuwata**
Follow-up for 010585873454d07625ee962ffa2ef2823624bfbe.

### CODE REVIEW

**Comment by yuwata** [comment]
@grawity @benjarobin I've tried to add detailed comments about what we actually do in the code. Does it make sense? Sorry, I am not native speaker. The explanations may be poor...

File: src/journal-remote/journal-remote.c [hunk 1]
```diff
@@ -664,7 +664,12 @@ static int setup_microhttpd_server(RemoteServer *s,
                 return log_error_errno(r, "Failed to make fd:%d nonblocking: %m", fd);
 
 /* MHD_OPTION_STRICT_FOR_CLIENT is introduced in microhttpd 0.9.54,
- *  and MHD_USE_PEDANTIC_CHECKS will be deprecated in future. */
+ * and MHD_USE_PEDANTIC_CHECKS will be deprecated in future.
+ * If MHD_USE_PEDANTIC_CHECKS is '#define'd, then it is deperecated
+ * and we should use MHD_OPTION_STRICT_FOR_CLIENT. On the other hand,
+ * if MHD_USE_PEDANTIC_CHECKS is not '#define'd, then it is not
+ * deperecated yet and there exists the same name enum element.
```
Comment (benjarobin) [inline src/journal-remote/journal-remote.c]: Spelling replace by: deprecated

File: src/journal-remote/journal-remote.c [hunk 2]
```diff
@@ -664,7 +664,12 @@ static int setup_microhttpd_server(RemoteServer *s,
                 return log_error_errno(r, "Failed to make fd:%d nonblocking: %m", fd);
 
 /* MHD_OPTION_STRICT_FOR_CLIENT is introduced in microhttpd 0.9.54,
- *  and MHD_USE_PEDANTIC_CHECKS will be deprecated in future. */
+ * and MHD_USE_PEDANTIC_CHECKS will be deprecated in future.
+ * If MHD_USE_PEDANTIC_CHECKS is '#define'd, then it is deperecated
```
Comment (benjarobin) [inline src/journal-remote/journal-remote.c]: Spelling replace by: deprecated

File: src/journal-remote/microhttpd-util.h [hunk 3]
```diff
@@ -24,12 +24,10 @@
...
-/* Those defines are added when options are renamed, hence the check for the *old* name. */
-
-/* Compatiblity with libmicrohttpd < 0.9.38 */
-#ifndef MHD_HTTP_NOT_ACCEPTABLE
-#  define MHD_HTTP_NOT_ACCEPTABLE MHD_HTTP_METHOD_NOT_ACCEPTABLE
-#endif
+/* Those defines are added when options are renamed. If the old names
+ * are not '#define'd, then they are not deperecated yet and there
+ * exist the enum elements with the same name. Hence let's check for
+ * the *old* name, and define the new name by the value the old name. */
```
Comment (benjarobin) [inline src/journal-remote/microhttpd-util.h]: by the value **of** the old name

File: src/journal-remote/microhttpd-util.h [hunk 4]
```diff
@@ -24,12 +24,10 @@
...
 #include "macro.h"
 
-/* Those defines are added when options are renamed, hence the check for the *old* name. */
-
-/* Compatiblity with libmicrohttpd < 0.9.38 */
-#ifndef MHD_HTTP_NOT_ACCEPTABLE
-#  define MHD_HTTP_NOT_ACCEPTABLE MHD_HTTP_METHOD_NOT_ACCEPTABLE
-#endif
+/* Those defines are added when options are renamed. If the old names
+ * are not '#define'd, then they are not deperecated yet and there
```
Comment (benjarobin) [inline src/journal-remote/microhttpd-util.h]: Spelling replace by: deprecated

File: src/journal-remote/microhttpd-util.h [hunk 5]
```diff
@@ -24,12 +24,10 @@
...
+ * exist the enum elements with the same name. Hence let's check for
```
Comment (benjarobin) [inline src/journal-remote/microhttpd-util.h]: Replace by: "and there are enum elements with the same name"

**Comment by yuwata** [comment]
@benjarobin Thank you. Updated.

**Comment by poettering** [comment]
CI failure appears unrelated

**Pull request merged**